keps/sig-node/24-apparmor/README.md

```diff
@@ -0,0 +1,790 @@
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [X] Feature gate 
+  - Feature gate name: `AppArmor`
+  - Components depending on the feature gate:
+      - kube-apiserver
+      - kubelet
+
+###### Does enabling the feature change any default behavior?
+  
+  N/A - the feature is already enabled by default since Kubernetes v1.4.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+  Yes, it works in the same way as before moving the feature to GA. However, the
+  GA related changes are backwards compatible, and the API supports rollback of
+  the Kubernetes API Server as described in the [Version Skew
+  Strategy](#version-skew-strategy).
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+  N/A - the feature is already enabled by default since Kubernetes v1.4.
+
+###### Are there any tests for feature enablement/disablement?
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+  N/A - the feature is already enabled by default since Kubernetes v1.4.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+  The [Version Skew Strategy](#version-skew-strategy) section covers this point.
+  Running workloads should have no impact as the Kubelet will support either the
+  existing annotations or the new fields introduced by this KEP.
+
+###### What specific metrics should inform a rollback?
+
+  N/A - the feature is already enabled by default since Kubernetes v1.4.
```

Comment:
default doesn't mean that it's on or must remain on. If a cluster that has disabled this wants to prepare for unconditional on, how would they know if they should revert?

Reply:
Or if the cluster-admin upgrades to this unconditionally on, what should he check to be sure that this feature is broken before he rolls back his upgrade?

Reply:
Addressed in [6de8b00](https://github.com/kubernetes/enhancements/pull/3298/commits/6de8b0031e0dfda6aeecab26a0c4e7fc91278de5)